Process_Library/GOCART2G_MieMod.F90

```diff
@@ -326,73 +330,88 @@ type(GOCART2G_Mie) function GOCART2G_MieCreate ( rcfile, wavelengths, nmom, rc )
       this%nMom = nmom_
       this%nPol = nPol_table
 
-      allocate (this%bext(this%nrh,this%nch,this%nbin), __NF_STAT__)
-      allocate (this%bsca(this%nrh,this%nch,this%nbin), __NF_STAT__)
-      allocate (this%bbck(this%nrh,this%nch,this%nbin), __NF_STAT__)
-      allocate (this%g(this%nrh,this%nch,this%nbin),    __NF_STAT__)
-      allocate (this%pback(this%nrh,this%nch,this%nbin,this%nPol),    __NF_STAT__)
-      if ( nmom_ > 0 ) then
-         allocate (this%pmom(this%nrh,this%nch,this%nbin,this%nMom,this%nPol),    __NF_STAT__)
-      end if
-      allocate (this%refr(this%nrh,this%nch,this%nbin), __NF_STAT__)
-      allocate (this%refi(this%nrh,this%nch,this%nbin), __NF_STAT__)
-
 !     Preserve the full RH structure of the input table
       this%rh = rh_table ! assignment does allocation
 
 !     Insert the requested channels in the output table
       this%wavelengths = wavelengths
 
-!     Insert rEff (moist effective radius)
-      this%reff = reff_table
+      allocate (this%gf(this%nbin),__NF_STAT__)
+      allocate (this%rEff(this%nbin),__NF_STAT__)
+      allocate (this%rhop(this%nbin),__NF_STAT__)
+      allocate (this%rhod(this%nbin),__NF_STAT__)
+      allocate (this%vol(this%nbin),__NF_STAT__)
+      allocate (this%area(this%nbin),__NF_STAT__)
+
+      do j = 1, this%nbin
+        this%reff(j)%rh = reff_table(:,j)
+        !Insert growth factor
+        this%gf(j)%rh   = gf_table(:,j)
+        !Wet particle density [kg m-3]
+        this%rhop(j)%rh = rhop_table(:,j)
+        !Dry particle density [kg m-3]
+        this%rhod(j)%rh = rhod_table(:,j)
+        !Volume [m3 kg-1]
+        this%vol(j)%rh  = vol_table(:,j)
+        !Area [m2 kg-1]
+        this%area(j)%rh = area_table(:,j)
+      enddo
```

Comment:
There should be a procedure that fully allocates each component of the derived type.  (Might need 2 or 3 such to handle the oddballs like pmom and rEff.)